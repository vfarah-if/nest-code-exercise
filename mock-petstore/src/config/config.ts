const config = {
  port: parseInt(process.env.PORT || '4444', 10),
  isDev: process.env.NODE_ENV === 'development',
  isProd: process.env.NODE_ENV === 'production',
  isTest: process.env.NODE_ENV === 'test',
  isInMemoryDatabase:
    process.env.DATABASE_IN_MEMORY?.toLowerCase() === 'true' || true,
  dbPort: parseInt(process.env.DB_PORT || '27017', 10),
  dbUrl: process.env.DATABASE_URL,
  dbName: process.env.DATABASE_DBNAME || 'petstore',
  dbUser: process.env.DATABASE_USER,
  dbPassword: process.env.DATABASE_PASS,
}
// console.debug('.env config', config)
export { config }
