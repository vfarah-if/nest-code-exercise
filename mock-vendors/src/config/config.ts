const DEFAULT_DB_PORT = '27017'
const DEFAULT_PORT = '4444'
const DEV_NODE_ENV = 'development'
const PROD_NODE_DEV = 'production'
const TEST_NODE_EV = 'test'
const DEFAULT_DB_NAME = 'vendors'
const config = {
  port: parseInt(process.env.PORT || DEFAULT_PORT, 10),
  isDev: process.env.NODE_ENV === DEV_NODE_ENV,
  isProd: process.env.NODE_ENV === PROD_NODE_DEV,
  isTest: process.env.NODE_ENV === TEST_NODE_EV,
  isInMemoryDatabase:
    process.env.DATABASE_IN_MEMORY?.toLowerCase() === 'true' || true,
  dbPort: parseInt(process.env.DB_PORT || DEFAULT_DB_PORT, 10),
  dbUrl: process.env.DATABASE_URL,
  dbName: process.env.DATABASE_DBNAME || DEFAULT_DB_NAME,
  dbUser: process.env.DATABASE_USER,
  dbPassword: process.env.DATABASE_PASS,
  hasAuthDetails:
    process.env.DATABASE_USER || process.env.DATABASE_PASS ? true : false,
  contentStackDeliveryApi: {
    environment: process.env.CONTENT_STACK_ENVIRONMENT,
  },
  version: process.env.VERSION || 'Unknown',
  sapApi: {
    jwt_secret: process.env.JWT_SECRET,
    jwt_expire_time: process.env.JWT_EXPIRE_TIME || '15m',
  },
}
console.debug('.env', config)
export { config }
