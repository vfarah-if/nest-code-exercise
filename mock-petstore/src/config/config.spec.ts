import { config } from './config'

describe('config', () => {
  beforeEach(() => {
    require('dotenv').config({
      path: '../../env.test',
    })
  })
  it('should configure defaults', () => {
    expect(config).toMatchSnapshot()
  })

  // DATABASE_URL="mongodb://127.0.0.1:27017/petstore?ssl=false&connectTimeoutMS=5000&maxPoolSize=50"
  // NODE_ENV='development'
  // DATABASE_PORT='27107'
  // DATABASE_IN_MEMORY=true
  // DATABASE_DBNAME='petstore'
  // DATABASE_USER=
  // DATABASE_PASS=
})
