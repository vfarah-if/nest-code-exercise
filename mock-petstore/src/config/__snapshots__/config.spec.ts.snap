// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config should configure defaults 1`] = `
Object {
  "dbName": "test-petstore",
  "dbPassword": "P@ssword1",
  "dbPort": 27017,
  "dbUrl": "mongodb://127.0.0.1:12345/test-petstore?ssl=false&connectTimeoutMS=5000&maxPoolSize=50",
  "dbUser": "user",
  "hasAuthDetails": true,
  "isDev": false,
  "isInMemoryDatabase": true,
  "isProd": false,
  "isTest": true,
  "port": 5555,
}
`;
